src/onnxruntime.cc

```diff
@@ -682,10 +682,40 @@ ModelState::AutoCompleteConfig()
   OrtAllocator* default_allocator;
   std::string model_path;
   {
+    TRITONSERVER_InstanceGroupKind kind;
+
+    triton::common::TritonJson::Value instance_group;
+    ModelConfig().Find("instance_group", &instance_group);
+
+    // Earlier in the model lifecycle, device checks for the instance group
+    // have already occurred. If at least one instance group with
+    // "kind" = "KIND_GPU" then allow model to use GPU else autocomplete to CPU
+    bool found_gpu_instance = false;
+    for (size_t i = 0; i < instance_group.ArraySize(); ++i) {
+      triton::common::TritonJson::Value instance_obj;
+      instance_group.IndexAsObject(i, &instance_obj);
+
+      triton::common::TritonJson::Value instance_group_kind;
+      instance_obj.Find("kind", &instance_group_kind);
+      std::string kind_str;
+      RETURN_IF_ERROR(instance_group_kind.AsString(&kind_str));
+
+      if (kind_str == "KIND_GPU" || kind_str == "KIND_AUTO") {
```

Comment:
Is it possible for `kind_str == "KIND_AUTO"`? I believe model_config_utils.cc in Triton core will override KIND_AUTO to the correct KIND_CPU/KIND_GPU.

Reply:
I believe if the bakcend gets KIND_AUTO from Triton Core here, then we must raise an error instead of taking it as GPU instance.

Reply:
I looked through the code again (https://github.com/triton-inference-server/core/blob/2ba05e79e8f5db7c354f2ffb48cc8cd920c23be7/src/model_config_utils.cc#L749). You are right, KIND_AUTO should be processed appropriately. I am taking out the "KIND_AUTO" check.